lib/single-new-eks-opensource-observability-pattern/gpu-index.ts

```diff
@@ -0,0 +1,121 @@
+
+        const account = process.env.COA_ACCOUNT_ID! || process.env.CDK_DEFAULT_ACCOUNT!;
+        const region = process.env.COA_AWS_REGION! || process.env.CDK_DEFAULT_REGION!;
+        const ampWorkspaceName = process.env.COA_AMP_WORKSPACE_NAME! || 'observability-amp-Workspace';
+        const ampWorkspace = blueprints.getNamedResource(ampWorkspaceName) as unknown as amp.CfnWorkspace;
+        const ampEndpoint = ampWorkspace.attrPrometheusEndpoint;
+        const ampWorkspaceArn = ampWorkspace.attrArn;
+        
+        const amgEndpointUrl = process.env.COA_AMG_ENDPOINT_URL;
+
+        // All Grafana Dashboard URLs from `cdk.json`
+        const fluxRepository: blueprints.FluxGitRepo = utils.valueFromContext(scope, "fluxRepository", undefined);
+        fluxRepository.values!.AMG_AWS_REGION = region;
+        fluxRepository.values!.AMP_ENDPOINT_URL = ampEndpoint;
+        fluxRepository.values!.AMG_ENDPOINT_URL = amgEndpointUrl;
+
+        const ampAddOnProps: blueprints.AmpAddOnProps = {
+            ampPrometheusEndpoint: ampEndpoint,
+            ampRules: {
+                ampWorkspaceArn: ampWorkspaceArn,
+                ruleFilePaths: [
+                    __dirname + '/../common/resources/amp-config/alerting-rules.yml',
+                    __dirname + '/../common/resources/amp-config/recording-rules.yml'
+                ]
+            }
+        };
+
+        const gpuNodeGroup: GpuNodeGroupProps = utils.valueFromContext(scope, "gpuNodeGroup", undefined);
+        if (gpuNodeGroup === undefined) {
+            throw new Error("Missing node group configuration");
+        }
+
+        Reflect.defineMetadata("ordered", true, blueprints.addons.GrafanaOperatorAddon);
+        const addOns: Array<blueprints.ClusterAddOn> = [
+            new blueprints.addons.CloudWatchLogsAddon({
+                logGroupPrefix: `/aws/eks/${stackId}`,
+                logRetentionDays: 30
+            }),
+            new blueprints.addons.XrayAdotAddOn(),
+            new blueprints.addons.FluxCDAddOn({"repositories": [fluxRepository]}),
+            new GrafanaOperatorSecretAddon(),
+            new blueprints.addons.VpcCniAddOn(),
+            new blueprints.addons.GpuOperatorAddon({
+                version: "v23.6.1",
+                values: {
+                    driver: {
+                        enabled: false // the GPU driver is already embedded into the AL2_X86_64_GPU AMI image 
+                    }
+                }
+            })
+        ];
+
+        ObservabilityBuilder.builder()
+            .account(account)
+            .region(region)
+            .version('auto')
+            .enableOpenSourcePatternAddOns(ampAddOnProps)
+            .resourceProvider(ampWorkspaceName, new blueprints.CreateAmpProvider(ampWorkspaceName, ampWorkspaceName))
+            .clusterProvider(
+                new blueprints.GenericClusterProvider({
+                    version: eks.KubernetesVersion.V1_27,
```

Comment:
Why is this required when. you are providing `version('auto')`

Reply:
right, removed

> Why is this required when. you are providing `version('auto')`

right, removed